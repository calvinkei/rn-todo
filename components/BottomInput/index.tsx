import React, {RefObject} from 'react';
import {
  Button,
  View,
  StyleSheet,
  TextInput,
  TextInputProps,
} from 'react-native';

interface BottomInputProps extends TextInputProps {
  onEnter: () => void;
  buttonTitle: string;
  inputRef?: RefObject<TextInput>;
}

function BottomInput({
  onEnter,
  buttonTitle,
  inputRef,
  ...props
}: BottomInputProps): JSX.Element {
  return (
    <View style={styles.container}>
      <TextInput ref={inputRef} style={styles.input} {...props} />
      <Button testID="enter button" title={buttonTitle} onPress={onEnter} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 12,
    borderTopWidth: 1,
    borderTopColor: '#ddd',
  },
  input: {
    flex: 1,
    backgroundColor: '#fff',
    borderRadius: 12,
    paddingHorizontal: 12,
    marginRight: 8,
  },
});

export default BottomInput;
