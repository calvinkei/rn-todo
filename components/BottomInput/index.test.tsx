import 'react-native';
import React from 'react';
import BottomInput from '.';
import {it, jest, expect} from '@jest/globals';
import renderer from 'react-test-renderer';

it('matches previous snapshot', () => {
  const onEnter = jest.fn();
  const tree = renderer
    .create(<BottomInput onEnter={onEnter} buttonTitle="title" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('calls onEnter when enter button is clicked', () => {
  const onEnter = jest.fn();
  const instance = renderer.create(
    <BottomInput onEnter={onEnter} buttonTitle="title" />,
  );
  instance.root.findByProps({testID: 'enter button'}).props.onPress();
  expect(onEnter).toBeCalledTimes(1);
});
