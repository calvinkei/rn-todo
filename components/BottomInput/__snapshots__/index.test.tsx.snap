// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "borderTopColor": "#ddd",
      "borderTopWidth": 1,
      "flexDirection": "row",
      "padding": 12,
    }
  }
>
  <TextInput
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 12,
        "flex": 1,
        "marginRight": 8,
        "paddingHorizontal": 12,
        "paddingVertical": 12,
      }
    }
  />
  <View
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="enter button"
  >
    <View
      style={
        [
          {},
        ]
      }
    >
      <Text
        style={
          [
            {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        title
      </Text>
    </View>
  </View>
</View>
`;
